model/task_lifecycle.go

```diff
@@ -956,7 +957,7 @@ func UpdateDisplayTask(t *task.Task) error {
 
 	t.Status = status
 	t.TimeTaken = timeTaken
-	if t.IsFinished() {
+	if t.IsFinished() && !wasFinished {
```

Comment:
Instead of this, can we just delete the event.LogDisplayTaskFinished call here? Reason being that right now there is a call to event.LogTaskFinished in MarkEnd, so every display task results in two task finished events being logged. 

Alternatively, can you change the key for display tasks finished logging from "TASK_FINISHED", so they don't clobber each other?

Reply:
MarkEnd is never called for a display task - this is what marks it as ended. I don't want to change the event key because task finish notifications should go out for this